### DISCUSSION TITLE
filesystem.renamer(): more checks on non-Windows (like on Windows)

### DISCUSSION THREAD

**Discussion by sanderjo** [sabnzbd/sabnzbd#2158] [Ideas]
filesystem.renamer() does checking and retrying on Windows, but not on non-Windows (Linux, Synology, MacOS). So my proposal is to do checking on non-Windows too.

Proposal:

- At the beginning, around https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/filesystem.py#L851-L853 , check that `old` exsist, and if not ... logging.error and `raise OSError() `
- https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/filesystem.py#L898 : put a try except around it. And in case of an exception, a logging.error and `raise OSError() `
- Maybe also do a few retries on non-Windows?

I don't know how the `raise OSError()` is then handled by SABnzbd? It is picked up nicely?

Triggered by https://forums.sabnzbd.org/viewtopic.php?p=127306#p127306

**Comment by Safihre** [comment]
The question is why, on Windows it makes sense to repeat because the sources of the problem (virusscanner, other programs) might resolve in the next try. But on Linux, the problems are usually persistent ones: permissions, disk space or disk unavailable. Retrying just delays inevitable failure.

**Comment by sanderjo** [comment]
That is the answer to suggestion 3, or all suggestions?

**Reply by Safihre** [reply]
To suggestion 3.

For the other 2 suggestions: they would result themselves in some kind of exception so why add extra code for it?

**Reply by sanderjo** [reply]
It would at least say to the user "Could not find original directory".

Now users getting the Traceback think it has to do with SABnzbd's renaming / deobfuscation, because the last message (before the Traceback) is "Renaming blabla to blabla". IMHO confusing.

**Reply by Safihre** [reply]
Oke, we could do that. You will make a PR?
